### ISSUE TITLE
Should Rule S101 understand acronyms like "2fa"?

### ISSUE THREAD

**Issue by Sjlver** [SonarSource/sonar-dotnet#5497]
### Description

Rule S101 currently generates a warning for the following code, which is generated by the [ASP.NET Core identity scaffolder](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity):

```csharp
public class LoginWith2faModel : PageModel
{
  // ...
}
```

```
warning S101: Rename class 'LoginWith2faModel' to match pascal case naming rules, consider using 'LoginWith2FaModel'.
```

### Expected behavior

I believe the generated code is correct here. "2fa" (short for *two-factor authentication*) is a common acronym and should be treated as a single "word" as far as pascal-case is concerned.

### Actual behavior

Rule S101 does not know about "2fa" and treats it like two words "2" and "fa".

### Related information

* SonarAnalyzer.CSharp version 8.35.0.42613

**Comment by costin-zaharia-sonarsource** [comment]
Duplicate of:
- https://github.com/SonarSource/sonar-dotnet/issues/2440
- https://github.com/SonarSource/sonar-dotnet/issues/320

**Comment by costin-zaharia-sonarsource** [comment]
I forgot to mention, that this rule already supports a `Pattern` parameter that is used to change the behavior. It accepts a regex string.
